src/lib/y2packager/repomd_downloader.rb

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2019 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "rexml/document"
+require "yast"
+
+Yast.import "Pkg"
+
+module Y2Packager
+  # This class downloads the primary.xml metadata file from an rpm-md repository.
+  # For multi-product repositories it downloads the files from all product
+  # subdirectories.
+  class RepomdDownloader
+    attr_reader :url, :product_repos
+
+    #
+    # Constructor
+    #
+    # @param url [String] URL of the repository
+    #
+    def initialize(url)
+      @url = url
+    end
+
+    #
+    # Download the primary.xml file(s) from the repository.
+    #
+    # @return [Array<String>] List of paths pointing to the downloaded primary.xml files,
+    #   returns an empty list if the URL or the repository is not valid.
+    #
+    def download
```

Comment:
name is a bit confusing, what about something like `primary_xmls` as what you want is actually that files. Also that side-effect for product_repos is not nice, as I mention above.

Reply:
and maybe caching of that primary_xmls? and also please document who is reponsible for deleting that files.

Reply:
Renamed, documented. I'd avoid caching as the files might be removed from the outside. If somebody needs caching it must be implemented on the caller side.